src/Utility/Xml.php

```diff
@@ -341,7 +343,14 @@ protected static function _fromArray(
                             // https://www.w3.org/TR/REC-xml/#syntax
                             // https://bugs.php.net/bug.php?id=36795
                             $child = $dom->createElement($key, '');
-                            $child->appendChild(new DOMText((string)$value));
+                            if ($value instanceof BackedEnum) {
+                                $value = (string)$value->value;
+                            } elseif ($value instanceof UnitEnum) {
```

Comment:
Handling `BackedEnum` is fine but I don't think we should be dealing with `UnitEnum` as they have no associated value. For reference you can `json_encode()` `BackedEnum` but not `UnitEnum`.

https://3v4l.org/6h7cm#v8.3.13

Reply:
Interesting.
I thought it doesn't hurt to add (as did we in other places), as the data can also come from any custom array that might include them.
So I don't see the harm of this check here for consistency - and to avoid it blowing up because of this. Hard to document that this is not supported for XML, but e.g. for Collection.

Reply:
I think including the name is a reasonable solution. If devs want to control the output more, they can use a `BackedEnum`.